### PR TITLE
API: do not return dnssec info in domain list

### PR DESCRIPTION
**Pull request by zeha**
### Short description

Stop backend creation in a tight loop when listing all domains over the API. Should improve performance considerably. Also notes in the docs that "list" is incomplete.
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled and tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added regression tests
- [ ] added unit tests

### CODE REVIEW

**Comment by Habbie** [comment]
How about moving the backend creation outside of the loop? Would at least allow db connection reuse.

**Comment by zeha** [comment]
Yes, but we'll still send 2+ SQL queries in a loop then...

**Comment by Habbie** [comment]
True. Maybe Wiebel can measure for us on his real load?

. o O ( ideally we would do all dnssec status in a single query )

**Comment by vandalon** [comment]
Just did. The old pdns (version 4.0.1+225) takes about 8.5 seconds. the same version with this patch less then 0.3 seconds.

**Comment by zeha** [comment]
Jeeves reports that nsedit uses this.

**Comment by tuxis-ie** [comment]
Perhaps it is a good idea to change https://github.com/PowerDNS/pdns/blob/master/pdns/dbdnsseckeeper.cc#L59 to:

```
  keyset_t keys = getKeys(zone, true); // does the cache
```

and actually use the mentioned cache?

**Comment by zeha** [comment]
> and actually use the mentioned cache?

this is actually the default with no param supplied.

**Comment by zengxibai** [comment]
We have a few thousands zones (in the table ‘domains') hosted in PowerDNS database (MySQL). it looks very slow (10-12 minutes) to get a list of all domains with API call below:

curl -H 'X-API-Key: mykey' 'http://127.0.0.1:8080/api/v1/servers/localhost/zones/'

Is this a noticed performance problem? Any insight where the slowness comes from? 

I do see the DNSSEC info in the returned answer. below is one of the returned zones info. e.g. 

  {
    "account": "",
    "dnssec": false,
    "id": "188.28.10.in-addr.arpa.",
    "kind": "Native",
    "last_check": 0,
    "masters": [],
    "name": "188.28.10.in-addr.arpa.",
    "notified_serial": 0,
    "serial": 2017081704,
    "url": "api/v1/servers/localhost/zones/188.28.10.in-addr.arpa."
  },

**Comment by zeha** [comment]
Rebased for 4.1.

**Comment by Habbie** [comment]
Can we make this a choice by the API client? `&dnssec=true` or the like?

**Comment by pieterlexis** [comment]
> We could, but if we flip the default to false, it's still a breaking change.

Please make the default `true` so we can merge this :)

**Comment by zeha** [comment]
Well, this got a bit harder because since 1258fec `edited_serial` is also returned in the zone list, which needs a DNSSECKeeper and does at least two metadata lookups and calls isSecuredZone. 

So either dnssec=false also turns off returning `edited_serial` or ... I dunno.

**Comment by wojas** [comment]
Turning off `edited_serial` is not ideal, but I would still be for merging this as is and documenting the behavior. The default stays the same, so this does not affect current users, only those that opt-in to the more efficient call.

**Comment by Habbie** [comment]
Agreed - let's add some docs and merge this.

**Comment by Habbie** [comment]
Testing and documenting this now.

**Comment by Habbie** [comment]
Documentation pushed, renders like this:

image

**Comment by Habbie** [comment]
> looks good, could use a test

Oh yes, I somehow imagined it had one already. Will do.

**Comment by Habbie** [comment]
Test pushed.

File: regression-tests.api/test_Zones.py [hunk 1]
```diff
@@ -59,13 +62,22 @@ def test_list_zones(self):
...
-            required_fields = required_fields + ['masters', 'last_check', 'notified_serial', 'edited_serial', 'serial', 'account']
+            required_fields = required_fields + ['masters', 'last_check', 'notified_serial', 'serial', 'account']
+            if dnssec:
+                required_fields = required_fields = ['dnssec', 'edited_serial']
             self.assertNotEquals(example_com['serial'], 0)
         elif is_recursor():
             required_fields = required_fields + ['recursion_desired', 'servers']
         for field in required_fields:
             self.assertIn(field, example_com)
+        if not dnssec:
```
Comment (pieterlexis) [inline regression-tests.api/test_Zones.py]: I think this should be inside the `if is_auth()` block

Comment (Habbie) [reply regression-tests.api/test_Zones.py]: Fixed, and now we also don't pointlessly test `dnssec=false` on the Recursor.

**Comment by Habbie** [comment]
Rebased.

**Pull request merged**